### ISSUE TITLE
Provide function for resampling profiles

### ISSUE THREAD

**Issue by kingsleynweye** [HichamJohra/energy_flexibility_kpis#15]
Some KPIs are time resolution agnostic and require the user to provide profiles in the resolution that they want KPIs to be calculated in. To help with profile preparation, the [energy_flexibility_kpis.utilities](https://github.com/HichamJohra/energy_flexibility_kpis/blob/develop/energy_flexibility_kpis/utilities.py) will have a resampling function.

**Comment by HichamJohra** [comment]
Perfect :) All those "support" function for data pre-treatment before direct use into KPIs function will thus be placed in "utilities" I guess

**Comment by kingsleynweye** [comment]
> Perfect :) All those "support" function for data pre-treatment before direct use into KPIs function will thus be placed in "utilities" I guess

Yes, that's the plan :).

**Comment by HichamJohra** [comment]
@kingsleynweye , @ZhelunChen : another nice set of "utilities" functions could some to check the quality of the input dataset. Something like a set of basic functions:
- give basic intel on the time series: mean, median, 10th/90th percentile, total length in time, sampling distribution/time step distribution, total number of data points, number of NaN
- quality data function: take "sampling rate" as input and check the number of missing data points according to that sampling rate, gives distribution of data gaps (occurrence data gaps as a function of length of data gaps), overall % of missing information.

this functions would give information to the user when uploading data before calculating KPIs. We could set some warning message for pre-defined data quality thresholds.

What do you think?

**Comment by kingsleynweye** [comment]
> @kingsleynweye , @ZhelunChen : another nice set of "utilities" functions could some to check the quality of the input dataset. Something like a set of basic functions:
> 
> * give basic intel on the time series: mean, median, 10th/90th percentile, total length in time, sampling distribution/time step distribution, total number of data points, number of NaN
> * quality data function: take "sampling rate" as input and check the number of missing data points according to that sampling rate, gives distribution of data gaps (occurrence data gaps as a function of length of data gaps), overall % of missing information.
> 
> this functions would give information to the user when uploading data before calculating KPIs. We could set some warning message for pre-defined data quality thresholds.
> 
> What do you think?

@HichamJohra yes, I agree with providing a utility functions that return summary statistics. I like the quality check and using that to provide warnings.

I'll create a new issue concerning this.
